Unit test flake: pkg/kvstore/store: "timeout while waiting for expected value"

## Unit test flake: pkg/kvstore/store: ""timeout while waiting for expected value"

Reproducible also locally in the dev VM when running "go test" in `pkg/kvstore/store` for a few times:

```
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
level=warning msg="Received delete event for local key. Re-creating the key in the kvstore" eventType=delete key=NmHJwKEsbeNd/local1 storeName=store-NmHJwKEsbeNd subsys=shared-store
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
level=warning msg="Received node delete event for node zjsUywFtJOUT-local1 which re-appeared within 30s" subsys=shared-store
----------------------------------------------------------------------
FAIL: <autogenerated>:1: StoreConsulSuite.TestStoreOperations
store_test.go:204:
    c.Assert(expect(func() bool { return localKey2.deleted() == 0 }), IsNil)
... value *errors.errorString = &errors.errorString{s:"timeout while waiting for expected value"} ("timeout while waiting for expected value")
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
OOPS: 9 passed, 1 FAILED
--- FAIL: Test (131.24s)
FAIL
coverage: 9.9% of statements in ./...
FAIL	github.com/cilium/cilium/pkg/kvstore/store	131.286s
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```
- Set title to be in the format `CI: <test-name>`
- Copy-paste the output the test failure
- Upload the zip file generated from that test failure
- Copy-paste the link of the CI build where that test failure has happen

--- a/pkg/kvstore/store/store_test.go
+++ b/pkg/kvstore/store/store_test.go
@@ -69,6 +69,7 @@ func (e *StoreConsulSuite) SetUpTest(c *C) {
 func (e *StoreConsulSuite) TearDownTest(c *C) {
 	kvstore.DeletePrefix(testPrefix)
 	kvstore.Close()
+	time.Sleep(defaults.NodeDeleteDelay + 5*time.Second)
 }
 
 type TestType struct {
@@ -198,7 +199,7 @@ func (s *StoreSuite) TestStoreOperations(c *C) {
 	c.Assert(expect(func() bool { return localKey3.updated() == 0 }), IsNil)
 
 	store.DeleteLocalKey(&localKey1)
-	// localKey2 will be deleted 2 times, one from local key and other from
+	// localKey1 will be deleted 2 times, one from local key and other from
 	// the kvstore watcher
 	c.Assert(expect(func() bool { return localKey1.deleted() == 2 }), IsNil)
 	c.Assert(expect(func() bool { return localKey2.deleted() == 0 }), IsNil)
@@ -242,8 +243,8 @@ func (s *StoreSuite) TestStorePeriodicSync(c *C) {
 	store.DeleteLocalKey(&localKey1)
 	store.DeleteLocalKey(&localKey2)
 
-	c.Assert(expect(func() bool { return localKey1.deleted() >= 1 }), IsNil)
-	c.Assert(expect(func() bool { return localKey2.deleted() >= 1 }), IsNil)
+	c.Assert(expect(func() bool { return localKey1.deleted() == 1 }), IsNil)
+	c.Assert(expect(func() bool { return localKey2.deleted() == 1 }), IsNil)
 }
 
 func (s *StoreSuite) TestStoreLocalKeyProtection(c *C) {

